packages/mobile/src/account/FiatExchange.tsx

```diff
@@ -0,0 +1,89 @@
+import colors from '@celo/react-components/styles/colors'
+import * as React from 'react'
+import { WithTranslation } from 'react-i18next'
+import { ActivityIndicator, StyleSheet } from 'react-native'
+import { WebView } from 'react-native-webview'
+import { connect } from 'react-redux'
+import { showError } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import i18n, { Namespaces, withTranslation } from 'src/i18n'
+import { LocalCurrencyCode } from 'src/localCurrency/consts'
+import { headerWithBackButton } from 'src/navigator/Headers'
+import { RootState } from 'src/redux/reducers'
+
+interface State {
+  signedUrl: string
+}
+
+interface StateProps {
+  localCurrency: LocalCurrencyCode
+  account: string | null
+}
+
+const mapStateToProps = (state: RootState): StateProps => {
+  return {
+    localCurrency: state.localCurrency.preferredCurrencyCode || LocalCurrencyCode.USD,
+    account: state.web3.account,
+  }
+}
+
+type Props = StateProps & WithTranslation
+
+const celoCurrencyCode = 'ETH' // TODO switch to cUSD when added to Moonpay
+const signMoonpayFirebaseUrl = 'https://us-central1-celo-org-mobile.cloudfunctions.net/signMoonpay'
+
+async function signMoonpayUrl(account: string, localCurrencyCode: LocalCurrencyCode) {
+  const response = await fetch(signMoonpayFirebaseUrl, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      currency: celoCurrencyCode,
+      address: account,
+      fiatCurrency: localCurrencyCode,
+    }),
+  })
+  const json = await response.json()
+  return json.url
+}
+
+class FiatExchange extends React.Component<Props, State> {
+  static navigationOptions = () => ({
+    ...headerWithBackButton,
+    headerTitle: i18n.t('accountScreen10:addFunds'),
+  })
+
+  state: State = {
+    signedUrl: '',
+  }
+
+  componentDidMount() {
+    if (this.props.account) {
+      signMoonpayUrl(this.props.account, this.props.localCurrency)
```

Comment:
Could you make this a separate function called in componentDidMount so that you can use async/await?